tests/test_entrypoint.py

```diff
@@ -0,0 +1,239 @@
+"""
+Testing the logic that extracts the entrypoint data:
+    - CLI entrypoints
+    - GUI entrypoints
+    - miscellaneous entrypoints
+"""
+
+from setuptools_pyproject_migration import WritePyproject
+from setuptools.dist import Distribution
+
+
+# Entry point parsing
+
+
+def test_parse_entrypoint_happypath():
+    """
+    Test we can extract the entry point name and target from a string.
+    """
+    assert WritePyproject._parse_entrypoint("ep=project.module:target") == ("ep", "project.module:target")
+
+
+def test_parse_entrypoint_happypath_whitespace():
+    """
+    Test we strip extraneous whitespace from the strings received.
+    """
+    assert WritePyproject._parse_entrypoint("  ep  =  project.module:target  ") == ("ep", "project.module:target")
+
+
+def test_parse_entrypoint_missing_eq():
+    """
+    Test that a string lacking a '=' is refused.
+    """
+    try:
+        WritePyproject._parse_entrypoint("ep project.module:target")
+        assert False, "Should not have been accepted as valid"
+    except ValueError as e:
+        assert str(e).endswith(" is not of the form 'name = module:function'")
```

Comment:
pytest has a [standard way to ensure that an exception gets raised](https://docs.pytest.org/en/7.3.x/reference/reference.html#pytest-raises):

```suggestion
    with pytest.raises(ValueError, match=" is not of the form 'name = module:function'$"):
        WritePyproject._parse_entrypoint("ep project.module:target")
```

(don't forget to `import pytest`)